hydromt/raster.py

```diff
@@ -553,37 +592,56 @@ def rotation(self) -> float:
 
         NOTE: rotated rasters with a negative dx are not supported.
         """
-        xs, ys = self.xcoords.data, self.ycoords.data
-        rot = 0
-        if xs.ndim == 2:
-            ddx1 = xs[0, -1] - xs[0, 0]
-            ddy1 = ys[0, -1] - ys[0, 0]
-            if not np.isclose(ddx1, 0):
-                rot = math.degrees(math.atan(ddy1 / ddx1))
-            else:
-                rot = -90
-            if ddx1 < 0:
-                rot = 180 + rot
-            elif ddy1 < 0:
-                rot = 360 + rot
+        if self._rotation is not None:  # cached rotation
+            return self._rotation
+        rot = None
+        if self._transform is not None:  # cached transform
+            # NOTE: it is not always possible to get the rotation from the transform
+            if self._transform.b == self._transform.d == 0:
```

Comment:
if transform.b or transform.d are floats and not an integers this is very error prone. In that case please use np.isclose